src/tenant/tests/test_admin.py

```diff
@@ -255,6 +282,149 @@ def test_cant_change_existing_name(self):
+
+    @override_settings(ROOT_URLCONF=__name__)
+    def test_delete_view(self):
+        """Delete view should restrict access and actually delete items, but leaves schemas in database."""
+        delete_dict = {"post": "yes", "confirmation": "owner/extra"}
+        delete_url = reverse("admin:tenant_tenant_delete", args=(self.extra_tenant.pk,))
+
+        # assert number of tenants, should be three objects (test, public and extra)
+        self.assertEqual(Tenant.objects.count(), 3)
+
+        # first case, access /admin/tenant/<pk>/delete/ page as anonymous user
+        # should returns 302 (login required)
+        response = self.client.get(delete_url)
+        self.assertEqual(response.status_code, 302)
+
+        # second case, add user should not be able to delete tenants
+        # should returns 403 (no permission)
+        self.client.force_login(self.adduser)
+        response = self.client.get(delete_url)
+        self.assertEqual(response.status_code, 403)
+        post = self.client.post(delete_url, delete_dict)
+        self.assertEqual(post.status_code, 403)
+        self.assertEqual(Tenant.objects.count(), 3)  # no changes
+        self.client.logout()
+
+        # third case, view user should not be able to delete tenants
+        # should returns 403 (no permission)
+        self.client.force_login(self.viewuser)
+        response = self.client.get(delete_url)
+        self.assertEqual(response.status_code, 403)
+        post = self.client.post(delete_url, delete_dict)
+        self.assertEqual(post.status_code, 403)
+        self.assertEqual(Tenant.objects.count(), 3)  # no changes
+        self.client.logout()
+
+        # fourth case, delete user can delete, but using incorrect confirmation keyword/phrase
+        # should returns 200 (same page, but with errors)
+        self.client.force_login(self.deleteuser)
+        response = self.client.get(delete_url)
+        self.assertEqual(response.status_code, 200)
+        post = self.client.post(delete_url, {"post": "yes", "confirmation": "stranger/something"})
+        self.assertEqual(post.status_code, 200)
+        self.assertEqual(Tenant.objects.count(), 3)  # no changes
+        self.assertContains(
+            post, "The confirmation does not match the keyword"
+        )
+
+        # fifth case, delete user can delete
+        # should returns 302 (redirect to admin homepage)
+        response = self.client.get(delete_url)
+        self.assertContains(response, "tenant/tenant/%s/" % self.extra_tenant.pk)
+        self.assertContains(response, "Summary")
+        self.assertContains(response, "Tenants: 1")
+        post = self.client.post(delete_url, delete_dict)
+        self.assertRedirects(post, reverse("admin:index"))
+        # tenant object was removed (extra tenant is gone)
+        self.assertEqual(Tenant.objects.count(), 2)
+        # ...but schema still in database
+        self.assertTrue(schema_exists("extra"))
```

Comment:
Can this be put in its own test?  This is critical, so better to have own explicit test that schema still exists after tenant deletion.

Reply:
yes, I've split the tests, see https://github.com/bytedeck/bytedeck/pull/1414/commits/c6b3fe11a1dcb90c0e3d5f0ed5cbd8f5f10e03a4